### PR TITLE
✨ Enhance table filters and error handling

### PR DESCRIPTION
**Pull request by RTLS**
This pull request enhances the table filtering functionality and improves error handling. Key changes include:

1. Updated the `Filter` type to include `columnAttnum` and `valueType` fields.
2. Added support for IS NULL and IS NOT NULL operators, disabling the value input when these are selected.
3. Consolidated `IntegerValue` and `FloatValue` into a single `NumberValue` type.
4. Improved datetime validation in the `DateTimeValue` changeset.
5. Enhanced error traversal and handling in the `Sequin.Error` module.
6. Added a `pg_simple_type_to_filter_type` function to map PostgreSQL types to filter types.
7. Updated the HTTP push consumer form to handle new filter structure and types.
8. Adjusted tests to reflect the changes in filter types and error handling.
9. Minor UI improvements, including updating the disabled input opacity.

These changes provide a more robust and flexible filtering system while improving the overall user experience and error feedback.

### CODE REVIEW

**Comment by RTLS** [comment]
* **#143** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/143?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: lib/sequin/consumers/source_table.ex [hunk 1]
```diff
@@ -25,6 +27,62 @@ defmodule Sequin.Consumers.SourceTable do
...
+        "<=" -> :<=
+        "in" -> :in
+        "not in" -> :not_in
+        "is null" -> :is_null
+        "not null" -> :not_null
+        _ -> raise "Invalid operator: #{external_operator}"
+      end
+    end
+
+    def from_external(%{
```
Comment (acco) [inline lib/sequin/consumers/source_table.ex:48]: good design choice

File: lib/sequin/consumers/source_table.ex [hunk 2]
```diff
@@ -25,6 +27,62 @@ defmodule Sequin.Consumers.SourceTable do
...
+          :not_in -> :list
+          _ -> value_type
+        end
+
+      value =
+        case value_type do
+          :list ->
+            (value || "")
+            |> String.split(",")
+            |> Enum.map(&String.trim/1)
```
Comment (acco) [inline lib/sequin/consumers/source_table.ex:70]: remove `[]` or `{}`? (or does trim do that)

File: lib/sequin/consumers/source_table.ex [hunk 3]
```diff
@@ -108,15 +165,15 @@ defmodule Sequin.Consumers.SourceTable.StringValue do
   end
 end
 
-defmodule Sequin.Consumers.SourceTable.IntegerValue do
+defmodule Sequin.Consumers.SourceTable.NumberValue do
```
Comment (acco) [inline lib/sequin/consumers/source_table.ex:168]: 👍

File: lib/sequin/consumers/source_table.ex [hunk 4]
```diff
@@ -126,40 +183,53 @@ defmodule Sequin.Consumers.SourceTable.IntegerValue do
...
+    end
   end
-end
-
-defmodule Sequin.Consumers.SourceTable.BooleanValue do
-  @moduledoc false
-  use Ecto.Schema
 
-  import Ecto.Changeset
+  @error_msg ~s(must be either "true" or "false")
```
Comment (acco) [inline lib/sequin/consumers/source_table.ex:211]:
```suggestion
  @error_msg ~s(must be either `true` or `false`)
```

? make it a little clearer they don't need quotes?

File: lib/sequin/consumers/source_table.ex [hunk 5]
```diff
@@ -126,40 +183,53 @@ defmodule Sequin.Consumers.SourceTable.IntegerValue do
...
-    struct
-    |> cast(params, [:value])
-    |> validate_required([:value])
+  defp validate_boolean_format(%{value: value}) when is_binary(value) do
+    case String.downcase(value) do
+      "true" -> {:ok, %{value: true}}
+      "false" -> {:ok, %{value: false}}
+      _ -> {:error, @error_msg}
+    end
   end
```
Comment (acco) [inline lib/sequin/consumers/source_table.ex:230]: I'm seeing double?

Comment (acco) [reply lib/sequin/consumers/source_table.ex:230]: Ah, I see - string vs atoms! Took me a minute -- I thought i was seeing double on these function headers. 

Maybe something like:

```
defp validate_boolean_format(%{value: value}), do: validate_boolean(value)
defp validate_boolean_format(%{"value" => value}), do: validate_boolean(value)

defp validate_boolean(value) when is_boolean(value) ...

File: lib/sequin/replication/message_handler.ex [hunk 6]
```diff
@@ -26,11 +28,15 @@ defmodule Sequin.Replication.MessageHandler do
 
   @impl MessageHandlerBehaviour
   def handle_messages(%Context{} = ctx, messages) do
+    Logger.info("[MessageHandler] Handling #{length(messages)} message(s)")
```
Comment (acco) [inline lib/sequin/replication/message_handler.ex:31]: just noting extra logs, fine w keeping while we debug

**Review by acco — APPROVED** [review]

**Pull request merged**